pkg/system/phase2_creating.go

```diff
@@ -1111,3 +1117,35 @@ func (r *Reconciler) SetEndpointsDeploymentReplicas(replicas int32) error {
 		return nil
 	})
 }
+
+// SetDesiredCoreAppConfig initiate the config map with predifined environment variables and their valuse
+func (r *Reconciler) SetDesiredCoreAppConfig() error {
+	updateNeeded := false
+	// When adding a new env var, make sure to add it to the sts/deployment as well
+	// see "NOOBAA_DISABLE_COMPRESSION" as an example
+	DefaultConfigMapData := map[string]string {
+		"NOOBAA_DISABLE_COMPRESSION": "false",
+		"DISABLE_DEV_RANDOM_SEED": "true",
+	}
+	for key, value := range DefaultConfigMapData {
+		if _, ok := r.CoreAppConfig.Data[key]; !ok {
+			r.CoreAppConfig.Data[key] = value
+			updateNeeded = true
+		}
+	}
+	if updateNeeded {
+		if !util.KubeUpdate(r.CoreAppConfig) {
+			r.Logger.Errorf("Failed to load initial data into %v", r.CoreAppConfig.Name)
+		}
+	}
+	return nil
+}
```

Comment:
there is no need to call `KubeUpdate` here. ReconcileObject will already update the object if needed. you just need to set the configmap with the desired values